src/data/finsupp/basic.lean

```diff
@@ -2220,6 +2220,36 @@ lemma sigma_sum [add_comm_monoid N] (f : (Σ (i : ι), αs i) → M → N) :
   l.sum f = ∑ i in split_support l, (split l i).sum (λ (a : αs i) b, f ⟨i, a⟩ b) :=
 by simp only [sum, sigma_support, sum_sigma, split_apply]
 
+variables {η : Type*} [fintype η] {ιs : η → Type*} [has_zero α]
+
+/-- On a `fintype η`, `finsupp.split` is an equivalence between `(Σ (j : η), ιs j) →₀ α`
+and `Π j, (ιs j →₀ α)`. -/
+noncomputable def sigma_finsupp_equiv_pi_finsupp :
+  ((Σ j, ιs j) →₀ α) ≃ Π j, (ιs j →₀ α) :=
+{ to_fun := λ f j, f.split j,
+  inv_fun := λ f, on_finset
+    (finset.sigma finset.univ (λ j, (f j).support))
+    (λ ji, f ji.1 ji.2)
+    (λ g hg, finset.mem_sigma.mpr ⟨finset.mem_univ _, mem_support_iff.mpr hg⟩),
+  left_inv := λ f, by { ext, simp [split] },
+  right_inv := λ f, by { ext, simp [split] } }
```

Comment:
If we don't have this already, can you add this for functions too? `((Σ j, ιs j) → α) ≃ Π j, (ιs j → α)

Reply:
Perhaps as:
```
noncomputable def equiv.arrow_sigma_equiv_pi_arrow :
  ((Σ j, ιs j) → α) ≃ Π j, (ιs j → α) :=
{ to_fun := λ f i x, f ⟨i, x⟩,
  inv_fun := λ f ij, f ij.1 ij.2,
  left_inv := λ f, funext $ λ ⟨i, j⟩, rfl,
  right_inv := λ f, rfl }
```

Reply:
Ah, this already exists as `equiv.Pi_curry`; so maybe just mention that in the docstring.